airflow/providers/apache/kafka/operators/consume_from_topic.py

```diff
@@ -0,0 +1,170 @@
+        confluent-python-kafka documentation), defaults to None, defaults to None
+    :param commit_cadence: When consumers should commit offsets ("never", "end_of_batch","end_of_operator"), defaults to "end_of_operator";
+         if end_of_operator, the commit() is called based on the max_messsages arg. Commits are made after the operator has processed the apply_function method for the maximum messages in the operator.
+         if end_of_batch, the commit() is called based on the max_batch_size arg. Commits are made after each batch has processed by the apply_function method for all messages in the batch.
+         if never,  close() is called without calling the commit() method.
+    :param max_messages: The maximum total number of messages an operator should read from Kafka,
+         defaults to None
+    :param max_batch_size: The maximum number of messages a consumer should read when polling, defaults to 1000
+    :param poll_timeout: How long the Kafka consumer should wait before determining no more messages are available,
+         defaults to 60
+    """
+
+    BLUE = "#ffefeb"
+    ui_color = BLUE
+    template_fields = (
+        "topics",
+        "apply_function",
+        "apply_function_args",
+        "apply_function_kwargs",
+        "config",
+    )
+
+    def __init__(
+        self,
+        topics: str | Sequence[str],
+        apply_function: Callable[..., Any] | str | None = None,
+        apply_function_batch: Callable[..., Any] | str | None = None,
+        apply_function_args: Sequence[Any] | None = None,
+        apply_function_kwargs: dict[Any, Any] | None = None,
+        consumer_config: dict[Any, Any] | None = None,
+        commit_cadence: str | None = "end_of_operator",
+        max_messages: int | None = None,
+        max_batch_size: int = 1000,
+        poll_timeout: float | None = 60,
+        **kwargs: Any,
+    ) -> None:
+
+        super().__init__(**kwargs)
+
+        self.topics = topics
+        self.apply_function = apply_function or None
+        self.apply_function_batch = apply_function_batch or None
+        self.apply_function_args = apply_function_args or ()
+        self.apply_function_kwargs = apply_function_kwargs or {}
+        self.config = consumer_config or {}
+        self.commit_cadence = commit_cadence
+        self.max_messages = max_messages or True
+        self.max_batch_size = max_batch_size
+        self.poll_timeout = poll_timeout
+
+        if self.commit_cadence not in VALID_COMMIT_CADENCE:
+            raise AirflowException(
+                f"commit_cadence must be one of {VALID_COMMIT_CADENCE}. Got {self.commit_cadence}"
+            )
+
+        if self.max_messages and self.max_batch_size > self.max_messages:
+            self.log.warning(
+                f"max_batch_size ({self.max_batch_size}) > max_messages"
+                f" ({self.max_messages}). Setting max_messages to"
+                f" {self.max_batch_size}"
+            )
```

Comment:
Let's avoid using f-string interpolation in logging statements and use %-formatting instead.

Reply:
looks like the pre-commit hook didn't catch this one.